docs/SIL.rst

```diff
@@ -636,40 +636,92 @@ Coroutine Types
 
 A coroutine is a function which can suspend itself and return control to
 its caller without terminating the function.  That is, it does not need to
-obey a strict stack discipline.
-
-SIL supports two kinds of coroutine: ``@yield_many`` and ``@yield_once``.
-Either of these attributes may be written before a function type to
-indicate that it is a coroutine type.
-
-A coroutine type may declare any number of *yielded values*, which is to
+obey a strict stack discipline. Different kinds of coroutines in SIL have
+different representations, reflecting the different capabilities and
+structural constraints of different language-level features:
```

Comment:
Does it make sense to call async functions a kind of coroutine?  They kind of aren't from SIL's perspective; they use mostly exactly the same SIL abstractions as normal functions.  Feels like it's just a new dimension of function type.

Reply:
Perhaps trying to discuss them as a kind of coroutine only serves to confuse. Thinking of them as just another axis of function type might be easier to understand; I'll try that.